ostd/src/mm/vm_space.rs

```diff
@@ -125,6 +125,11 @@ impl VmSpace {
         }
     }
 
+    /// Get an opaque identity that uniquely refers to a particular VmSpace instance.
+    pub fn id(&self) -> u64 {
+        core::ptr::addr_of!(self) as u64
+    }
```

Comment:
This is tricky because:
* unless an object is pinned, it's ID could change.
* a new object can have the same ID as an old one.
I really want a way to give things IDs, but I think this may not be the way to do it.

Do you need stability and uniqueness? If so, then lets fix it in this PR by using the something else if we can. If it's too hard to fix, let's document it and return to it later. I think we need a generic way to give things IDs, it's going to come up a lot.